gammapy/analysis/config.py

```diff
@@ -66,127 +79,212 @@ class Config:
+class DatasetsConfig(GammapyBaseConfig):
+    type: ReductionTypeEnum = ReductionTypeEnum.spectrum
     stack: bool = True
-    geom: Geom = Geom()
-    background: Background = Background()
-    onregion: SpatialCircleRange = SpatialCircleRange()
+    geom: GeomConfig = GeomConfig()
+    background: BackgroundConfig = BackgroundConfig()
+    on_region: SpatialCircleRangeConfig = SpatialCircleRangeConfig()
     containment_correction: bool = True
     psf_kernel_radius: AngleType = "0.6 deg"
 
 
-class Data(GammapyBaseModel):
+class ObservationsConfig(GammapyBaseConfig):
     datastore: Path = Path("$GAMMAPY_DATA/hess-dl3-dr1/")
     obs_ids: List[int] = []
     obs_file: FilePath = None
-    obs_cone: SpatialCircleRange = SpatialCircleRange()
-    obs_time: TimeRange = TimeRange()
+    obs_cone: SpatialCircleRangeConfig = SpatialCircleRangeConfig()
+    obs_time: TimeRangeConfig = TimeRangeConfig()
 
 
-class Log(GammapyBaseModel):
+class LogConfig(GammapyBaseConfig):
     level: str = "info"
     filename: Path = None
     filemode: str = None
     format: str = None
     datefmt: str = None
 
 
-class General(GammapyBaseModel):
-    log: Log = Log()
+class GeneralConfig(GammapyBaseConfig):
+    log: LogConfig = LogConfig()
     outdir: str = "."
 
 
-class AnalysisConfig(GammapyBaseModel):
+class AnalysisConfig(GammapyBaseConfig):
     """Config class handling the high-level interface settings."""
 
-    general: General = General()
-    data: Data = Data()
-    datasets: Datasets = Datasets()
-    fit: Fit = Fit()
-    flux_points: FluxPoints = FluxPoints()
+    general: GeneralConfig = GeneralConfig()
+    observations: ObservationsConfig = ObservationsConfig()
+    datasets: DatasetsConfig = DatasetsConfig()
+    fit: FitConfig = FitConfig()
+    flux_points: FluxPointsConfig = FluxPointsConfig()
+
+    @classmethod
+    def from_template(cls, template):
+        """Create AnalysisConfig from existing templates.
+
+        Parameters
+        ----------
+        template : {"1d", "3d"}
+            Built-in templates.
+
+        Returns
+        -------
+        config : `AnalysisConfig`
+            AnalysisConfig class
+        """
+        filename = CONFIG_PATH / ANALYSIS_TEMPLATES[template]
+        return cls.read(filename)
+
+    def __str__(self):
+        """Display settings in pretty YAML format."""
+        info = self.__class__.__name__ + "\n\n\t"
+
+        data = self.to_yaml()
+        data = data.replace("\n", "\n\t")
+        info += data
+        return info.expandtabs(tabsize=4)
+
+    @classmethod
+    def read(cls, filename):
+        config = read_yaml(filename)
+        return cls.from_yaml(config)
+
+    @classmethod
+    def from_yaml(cls, settings):
+        if isinstance(settings, str):
+            settings = yaml.safe_load(settings)
+        return AnalysisConfig(**settings)
+
+    def write(self, filename, overwrite=False):
+        fname = Path(filename).name
+        fpath = Path(self.general.outdir) / fname
```

Comment:
Suggest to just write to `filename`, not `outdir/filename`. Keep it simple. outdir is something that Analysis uses to put files.

I'm 60:40 on this choice, if you prefer, OK to keep `outdir/filename`.

But in any case, you have to add a docstring saying what this does (especially if it's `outdir/filename`, as a user I wouldn't expect that).

Reply:
Done